src/network/networkd-link.c

```diff
@@ -59,6 +59,19 @@ DUID* link_get_duid(Link *link) {
                 return &link->manager->duid;
 }
 
+static bool link_sysctl_ipv6_enabled(Link *link) {
+        if (link->sysctl_ipv6_enabled < 0) {
+                _cleanup_free_ char *value = NULL;
+
+                if (sysctl_read_ip_property(AF_INET6, link->ifname, "disable_ipv6", &value) >= 0)
+                        link->sysctl_ipv6_enabled = value[0] == '0';
+                else
+                        link->sysctl_ipv6_enabled = false;
+        }
```

Comment:
hmm, this is weird, as we managae this sysctl ourselves, see link_enable_ipv6. We shouldn't create a feedback loop here I think.

I am pretty sure we should manage the sysctl, and thus not read it. i.e. we should manage an interface as a whole or not at all, but design things so that an interface is partly configured by other stuff and partly by us. That said, in order to address #11711 I think it might make sense to read the global settings instead, and maybe readjust our behaviour based on that. i.e. when somebody used the global sysctls we'll not turn ipv6 on on any iface, even if .network says so.

Reply:
> That said, in order to address #11711 I think it might make sense to read the global settings instead, and maybe readjust our behaviour based on that.

Sounds good. I will update so.